cmd/gen-go-sample/main.go

```diff
@@ -81,60 +82,58 @@ func main() {
 		donec <- struct{}{}
 	}()
 
+	// TODO(hzyi): Read sample config
+
 	if err := os.MkdirAll(*outDir, 0755); err != nil {
 		log.Fatal(err)
 	}
 
 	<-donec
 	<-donec
 
-	for _, iface := range gen.gapic.Interfaces {
-		for _, meth := range iface.Methods {
-			if err := genMethodSamples(&gen, iface, meth, *nofmt, *outDir); err != nil {
-				err = errors.E(err, "generating: %s", iface.Name+"."+meth.Name)
-				log.Fatal(err)
-			}
-		}
-	}
-}
-
-func genMethodSamples(gen *generator, iface GAPICInterface, meth GAPICMethod, nofmt bool, outDir string) error {
-	valSets := map[string]SampleValueSet{}
-	for _, vs := range meth.SampleValueSets {
-		valSets[vs.ID] = vs
+	if err := genMethodSamples(&gen, gen.sampleConfig, *nofmt, *outDir); err != nil {
+		log.Fatal(err)
 	}
 
-	for _, sam := range meth.Samples.Standalone {
-		for _, vsID := range sam.ValueSets {
-			vs, ok := valSets[vsID]
-			if !ok {
-				return errors.E(nil, "value set not found: %q", vsID)
-			}
+}
 
-			gen.reset()
-			if err := gen.genSample(iface.Name, meth, sam.RegionTag, vs); err != nil {
-				return errors.E(err, "value set: %s", vsID)
-			}
-			content, err := gen.commit(!nofmt, time.Now().Year())
-			if err != nil {
-				return err
+func genMethodSamples(gen *generator, sampConf schema_v1p2.SampleConfig, nofmt bool, outDir string) error {
+	for _, samp := range sampConf.Samples {
+
+		for _, iface := range gen.gapic.Interfaces {
+			if iface.Name == samp.Service {
```

Comment:
I think it would be better to invert this to keep the body of this `if` statement less indented.

```go
if iface.Name != samp.Service {
  continue
}
```

Reply:
Done